eo-runtime/src/main/java/EOorg/EOeolang/EOrust.java

```diff
@@ -213,17 +220,28 @@ private static ConcurrentHashMap<String, String> load(final String src) throws I
      *  It must convert message array from 1 to last byte to the String
      *  and return eo object with converted String Data.
      */
-    private static Phi translate(final byte[] message) {
+    private Phi translate(final byte[] message) {
         final byte determinant = message[0];
         final byte[] content = Arrays.copyOfRange(message, 1, message.length);
         final Phi ret;
         switch (determinant) {
             case 0:
-                throw new ExFailure(
-                    "Returning vertex is not implemented yet"
-                );
+                ByteBuffer buffer = ByteBuffer.allocate(Integer.BYTES);
+                buffer.put(content);
+                buffer.flip();
+                final int vertex = buffer.getInt();
+                ret = this.phis.get(vertex);
+                if (ret == null) {
+                    throw new ExFailure(
+                        String.format(
+                            "Returned phi with vertex %d was not indexed",
+                            vertex
```

Comment:
@levBagryansky I'm not sure the vertex id is a good choice to error description. Can we provide some more details to this message?

Reply:
@Graur May be we can also print phis map ?

Reply:
@levBagryansky To print all of the map may cost a lot and may confuse the developer. Maybe we can print `content` as well?